tools/ci_connector_ops/ci_connector_ops/pipelines/publish.py

```diff
@@ -0,0 +1,232 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+import json
+import os
+import uuid
+from abc import ABC
+from typing import List, Tuple
+
+import anyio
+import dagger
+from ci_connector_ops.pipelines import builds
+from ci_connector_ops.pipelines.actions import environments
+from ci_connector_ops.pipelines.actions.remote_storage import upload_to_gcs
+from ci_connector_ops.pipelines.bases import ConnectorReport, Step, StepResult, StepStatus
+from ci_connector_ops.pipelines.contexts import ConnectorContext
+from ci_connector_ops.pipelines.pipelines import metadata
+from ci_connector_ops.pipelines.utils import with_stderr, with_stdout
+from dagger import Container, File, QueryError, Secret
+
+
+class PublishStep(Step, ABC):
+    def __init__(self, context: ConnectorContext, pre_release: bool = True) -> None:
+        super().__init__(context)
+        self.pre_release = pre_release
+
+    @property
+    def docker_image_name(self):
+        if self.pre_release:
+            return f"{self.context.docker_image_from_metadata}-dev.{self.context.git_revision[:10]}"
+        else:
+            return self.context.docker_image_from_metadata
+
+
+class CheckConnectorImageDoesNotExist(PublishStep):
+    title = "Check if the connector docker image does not exist on the registry."
+
+    async def _run(self) -> StepResult:
+        manifest_inspect = (
+            environments.with_docker_cli(self.context)
+            .with_env_variable("CACHEBUSTER", str(uuid.uuid4()))
+            .with_exec(["docker", "manifest", "inspect", self.docker_image_name])
+        )
+        manifest_inspect_stderr = await with_stderr(manifest_inspect)
+        manifest_inspect_stdout = await with_stdout(manifest_inspect)
+
+        if "no such manifest" in manifest_inspect_stderr:
```

Comment:
Any way to do this on an error code? or a message we define.

For example
```py
our_error_message = "SUPERUNIQUEERROR"
manifest_inspect = (
            environments.with_docker_cli(self.context)
            .with_env_variable("CACHEBUSTER", str(uuid.uuid4()))
            .with_exec(["docker", "manifest", "inspect", self.docker_image_name, "&&", "echo", our_error_message])
        )
        manifest_inspect_stderr = await with_stderr(manifest_inspect)
        manifest_inspect_stdout = await with_stdout(manifest_inspect)

        if our_error_message in manifest_inspect_stderr:
```

Definitely a hack, would still prefer an error code if possible